xbmc/Application.cpp

```diff
@@ -2778,6 +2779,27 @@ bool CApplication::OnMessage(CGUIMessage& message)
     }
     break;
 
+    case GUI_MSG_PLAY_TRAILER:
+    {
+      const CFileItemPtr item = std::static_pointer_cast<CFileItem>(message.GetItem());
+      std::unique_ptr<CFileItem> trailerItem =
+          ContentUtils::GeneratePlayableTrailerItem(*item.get(), g_localizeStrings.Get(20410));
```

Comment:
No need for a shared_ptr here at all:

```c++
       const CFileItem* item = std::static_pointer_cast<CFileItem>(message.GetItem());
       std::unique_ptr<CFileItem> trailerItem =
           ContentUtils::GeneratePlayableTrailerItem(*item, g_localizeStrings.Get(20410));
```

Reply:
We would have to delete the raw pointer manually, which is not what we want. So forget about my last comment.

But the shared_ptr could be changed to a unique_ptr. The latter is "cheaper".

Reply:
Huh, now I understand!

=> `CGUIListItemPtr CGUIMessage::GetItem() const`

We do an upcast here! CGUIListItem is the base class of CFileItem.

Then, the right approach is something like this:

```C++
const CFileItem* item = dynamic_cast<CFileItem>(message.GetItem().get());
if (item)
{
  std::unique_ptr<CFileItem> trailerItem =
      ContentUtils::GeneratePlayableTrailerItem(*item, g_localizeStrings.Get(20410));
```